web/src/utils/factory-management/dependencies.ts

```diff
@@ -97,15 +97,11 @@ export const calculateDependencyMetrics = (factory: Factory) => {
       const part = request.part
       const metrics = factory.dependencies.metrics
 
-      if (!factory.surplus[part]) {
-        console.warn(`calculateDependencyMetrics: Could not find a surplus for part ${part}.`, part)
-      }
-
       if (!metrics[part]) {
         metrics[part] = {
           part,
           request: 0,
-          supply: factory.surplus[part]?.amount ?? 0,
+          supply: factory.parts[part].amountRemaining ?? 0,
```

Comment:
This removes a circular dependency between export data and dependency metrics data, by using the part data as the source, which both of them use anyway.